index.ts

```diff
@@ -1,23 +1,100 @@
+import minimist from "minimist";
 import { delay, Logger, winston } from "./src/utils";
-
 import { runRelayer } from "./src/relayer";
 import { runDataworker } from "./src/dataworker";
 import { runMonitor } from "./src/monitor";
 import { runFinalizer } from "./src/finalizer";
 
 let logger: winston.Logger;
 
-export async function run(): Promise<void> {
+function usage(badInput: string = undefined) {
+  let usageStr = badInput ? `\nUnrecognized input: "${badInput}".\n\n` : "";
+  const userModes = "monitor|relayer";
+  const proModes = "dataworker|finalizer";
+  const walletOpts = "mnemonic|privateKey|gckms";
+
+  usageStr += `
+    Usage:
+    \tts-node ./index.ts --help
+    \tts-node ./index.ts [-h] <--monitor|--relayer>      --wallet <${walletOpts}>
+    \tts-node ./index.tx [-h] <--dataworker|--finalizer> --wallet <${walletOpts}>
+  `.slice(1);
+
+  console.log(usageStr);
+
+  // eslint-disable-next-line no-process-exit
+  process.exit(badInput === undefined ? 0 : -1);
+}
+
+function help() {
+  const botRepoUrl = "https://github.com/across-protocol/relayer-v2";
+  const relayerDocsUrl = "https://docs.across.to/v2/developers/running-a-relayer";
+  const helpStr = `
+    Across v2 Bot
+
+    Description:
+    \tThis application performs off-chain duties in support of the Across v2 bridge.
+
+    \tBasic sub-functions include:
+    \t  Monitor:    Monitor all relay and bundle events.
+    \t  Relayer:    Perform transaction relays based on eligible SpokePool deposits.
+
+    \tAdvanced sub-functions include:
+    \t  Dataworker: Monitor and produce relay bundles for the Hub Pool.
+    \t  Finalizer:  Finalize and collect canonical bridge transfers.
+
+    Links:
+    \tRepository: ${botRepoUrl}
+    \tRelayer Instructions: ${relayerDocsUrl}
+  `.slice(0, -1);
+
+  console.log(helpStr);
+  usage(); // no return
+}
+
+export async function run(args: { [k: string]: boolean | string }): Promise<void> {
   logger = Logger;
-  if (process.argv.includes("--relayer")) await runRelayer(Logger);
-  else if (process.argv.includes("--dataworker")) await runDataworker(Logger);
-  else if (process.argv.includes("--finalizer")) await runFinalizer(Logger);
-  else if (process.argv.includes("--monitor")) await runMonitor(Logger);
-  else console.log("Select either relayer, dataworker, or monitor to run!");
+
+  const cmds = {
+    dataworker: runDataworker,
+    finalizer: runFinalizer,
+    help: help,
+    monitor: runMonitor,
+    relayer: runRelayer,
+  };
+
+  /* XXX Make the mode of operation an operand, rather than an option.
+   * i.e. ts-node ./index.ts [options] <relayer|...>
+   */
+  let cmd;
+  for (const cmdName of Object.keys(cmds)) {
+    if (args[cmdName]) {
+      cmd = cmdName;
+      break;
+    }
+  }
```

Comment:
nit: this is a more idiomatic way to write this code. See find docs [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find).

```suggestion
  const cmd = Object.keys(cmds).find(cmdName => !!args[cmdName]);
```